### PR TITLE
New event page

### PR DESCRIPTION
**Pull request by kshepard**

### CODE REVIEW

File: web/test/spec/views/home/home-directive.spec.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    it('should load directive', function () {
+        var requestUrl = /\/api\/recordtypes/;
+        $httpBackend.expectGET(requestUrl).respond(200, ResourcesMock.RecordTypeResponse);
+
+        var scope = $rootScope.$new();
+        var element = $compile('<driver-home></driver-home>')(scope);
+        $rootScope.$apply();
+
+        expect(element.find('.form-area-body').length).toEqual(1);
```
Comment (ddohler) [inline web/test/spec/views/home/home-directive.spec.js]: Does there need to be a call to `$httpBackend.flush()` here?

File: web/test/spec/views/record/add-controller.spec.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            },
+            Person: [{
+                'abc': '123',
+                _localId: testUuid
+            }]
+        }, [], editor);
+
+        // Empty _localId fields should be filled in with uuids
+        expect(editorValue).toEqual(jasmine.any(Object));
+        expect(editorValue.Details._localId.length).toBe(36);
```
Comment (ddohler) [inline web/test/spec/views/record/add-controller.spec.js]: There's a UUID-matching regex in `builder-schema.json` that could allow this test to be a bit stricter.

File: web/test/spec/views/record/add-directive.spec.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    it('should load directive', function () {
+        var requestUrl = /\/api\/recordtypes/;
+        $httpBackend.expectGET(requestUrl).respond(200, ResourcesMock.RecordTypeResponse);
+
+        var scope = $rootScope.$new();
+        var element = $compile('<driver-record-add></driver-record-add>')(scope);
+        $rootScope.$apply();
+
+        expect(element.find('json-editor').length).toEqual(1);
```
Comment (ddohler) [inline web/test/spec/views/record/add-directive.spec.js]: Same question about `flush()` here although I think I may be missing something about when that call is and is not required.

**Comment by ddohler** [comment]
+1

File: web/test/spec/views/record/add-controller.spec.js (hunk 2 above)
Comment (kshepard) [inline web/test/spec/views/record/add-controller.spec.js]: Oh, cool, good point.

File: web/test/spec/views/home/home-directive.spec.js (hunk 1 above)
Comment (kshepard) [inline web/test/spec/views/home/home-directive.spec.js]: You may be right. I'll take a look into it and give it a shot.

**Pull request merged**